pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -62,6 +62,14 @@ func newPodCache(c *Controller, informer coreinformers.PodInformer, queueEndpoin
 	return out
 }
 
+func isPodReady(pod *v1.Pod) bool {
```

Comment:
Can you copy the k8s functions with a desc it is copied there https://github.com/kubernetes/kubernetes/blob/87984d84d18c7816835e5efedebb17ed13aa2509/pkg/api/v1/pod/util.go#L286-L289